File: tests/samples/javascript/class_with_methods.js
/**
 * A simple class with methods
 */
class MyClass {
    constructor() {
        this.value = 42;
    }
    
    /**
     * First method that returns a string
     */
    method1() {
        return "Hello from method1";
    }
    
    /**
     * Second method with parameters
     */
    method2(param1, param2 = 0) {
        return param1 + param2;
    }
}

/**
 * An arrow function example
 */
const arrowFunction = (param) => {
    return param * 2;
}; 

File: tests/samples/python/class_with_methods.py
class MyClass:
    """A simple class with methods."""
    
    def __init__(self, value=0):
        self.value = value
    
    def method1(self):
        """First method that returns a string."""
        return "Hello from method1"
        
    def method2(self, arg1, arg2=None):
        """Second method with multiple arguments.
        
        Args:
            arg1: First argument
            arg2: Optional second argument
        """
        if arg2 is None:
            return arg1
        return arg1 + arg2 